index.js

```diff
@@ -40,6 +40,22 @@ var decode = function(str) {
   }
 }
 
+/**
+ * Ensure any potential prototype pollution can be sanitized.
+ *
+ * @param {String} str
+ * @return {String}
+ * @api public
+ */
+
+var sanitizeObjKey = function(str) {
+  if (str && ["__proto__", "constructor", "prototype"].indexOf(str.toLowerCase()) > -1) {
+    return str.toUpperCase()
```

Comment:
Why convert to uppercase? If the user (for some weird reason) wanted the value of `prototype`, they'd get `PROTOTYPE` instead, which would either return unexpected (incorrect) results or `undefined`. Consider the following:

```js
var q = querystring.parse('prototype=hi&PROTOTYPE=bye')
console.log(q.prototype)
// => bye
console.log(q.PROTOTYPE)
// => bye
```

IMO we should either error or return `undefined`.